clab/dependency_manager/dependency_manager.go

```diff
@@ -2,101 +2,30 @@ package dependency_manager
 
 import (
 	"fmt"
+	"sort"
 	"strings"
-	"sync"
 
 	log "github.com/sirupsen/logrus"
+	"github.com/srl-labs/containerlab/types"
 )
 
 type DependencyManager interface {
 	// AddNode adds a node to the dependency manager.
 	AddNode(name string)
 	// AddDependency adds a dependency between depender and dependee.
 	// The depender will effectively wait for the dependee to finish.
-	AddDependency(dependee, depender string) error
+	AddDependency(depender string, dependerState types.WaitForPhase, dependee string, dependeeState types.WaitForPhase) error
 	// WaitForNodeDependencies is called by a node that is meant to be created.
 	// This call will bock until all the nodes that this node depends on are created.
-	WaitForNodeDependencies(nodeName string) error
+	Enter(nodeName string, state types.WaitForPhase) error
```

Comment:
what does `Enter` do? The comment is from the old func

Reply:
A Node calls Enter with it's own name and the Phase it tries to Enter... Blocking if there are dependencies that are not met yet.
Considered it a better name then WaitForNodeDependencies. This should also change in the next phase and become a Node bound private func that would not take the name any more.
Will change the comment.

Reply:
Done